### PR TITLE
cgroup: support mount of cgroup2

### PR DESCRIPTION
**Pull request by giuseppe**
convert a "cgroup" mount to "cgroup2" when the system uses cgroups v2
unified hierarchy.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
@filbranden FYI

**Comment by giuseppe** [comment]
tested with:
```
# podman run --cgroupns private -ti --rm --memory 100M fedora cat /sys/fs/cgroup/memory.max
104857600

# podman run -ti --rm --memory 100M fedora sh -c 'cat /sys/fs/cgroup/`sed -e "s|0::||g" /proc/self/cgroup`/memory.max'
104857600
```

File: libcontainer/rootfs_linux.go [hunk 1]
```diff
@@ -209,6 +209,74 @@ func prepareBindMount(m *configs.Mount, rootfs string) error {
...
+			if err := os.Symlink(mc, filepath.Join(rootfs, m.Destination, ss)); err != nil && !os.IsExist(err) {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func mountCgroupV2(m *configs.Mount, rootfs, mountLabel string, enableCgroupns bool) error {
+	cgroupPath := filepath.Join(rootfs, m.Destination)
```
Comment (TomSweeneyRedHat) [inline libcontainer/rootfs_linux.go]: any worries about symlinks here?

Comment (giuseppe) [reply libcontainer/rootfs_linux.go]: good point.  I've pushed a new version that uses `securejoin`

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**